server/src/main/java/org/elasticsearch/index/translog/BufferedChecksumStreamInput.java

```diff
@@ -70,6 +70,30 @@ public void readBytes(byte[] b, int offset, int len) throws IOException {
         digest.update(b, offset, len);
     }
 
+    private static final ThreadLocal<byte[]> buffer = ThreadLocal.withInitial(() -> new byte[8]);
+
+    @Override
+    public short readShort() throws IOException {
+        final byte[] buf = buffer.get();
```

Comment:
I think even for `short` this is better than the previous solution. It certainly is when reading from a Netty `ByteBuf` backed buffer since we save one round of bounds checks.